### PR TITLE
Writing bound and Solution docstrings 

### PR DESCRIPTION
**Pull request by ItamarRocha**
- [x] Create all the docstrings
- [x] Explain what each function is responsible for
- [x] Explain each functions args
- [x] check if the Documenter.jl is working fine

### CODE REVIEW

File: src/ColunaBase/solsandbounds.jl [hunk 1]
```diff
@@ -29,17 +29,18 @@ getvalue(b::Bound) = b.value
 Base.float(b::Bound) = b.value
 
 """
-    isbetter
+    isbetter(b1, b2)
 
-doc todo
+Returns true if bound b1 is better than bound b2.
+The function take into account the type of the bounds(Primal or Dual) and the Sense (MinSense or MaxSense).
```
Comment (guimarqu) [inline src/ColunaBase/solsandbounds.jl:35]:
```suggestion
The function take into account the space (primal or dual) and the objective sense (min, max) of the bounds.
```

File: src/ColunaBase/solsandbounds.jl [hunk 2]
```diff
@@ -29,17 +29,18 @@ getvalue(b::Bound) = b.value
...
 isbetter(b1::Bound{Sp,Se}, b2::Bound{Sp,Se}) where {Sp<:Primal,Se<:MaxSense} = b1.value > b2.value
 isbetter(b1::Bound{Sp,Se}, b2::Bound{Sp,Se}) where {Sp<:Dual,Se<:MinSense} = b1.value > b2.value
 isbetter(b1::Bound{Sp,Se}, b2::Bound{Sp,Se}) where {Sp<:Dual,Se<:MaxSense} = b1.value < b2.value
 
 """
-    diff 
+    diff
 
 distance to reach the dual bound from the primal bound;
 non-positive if dual bound reached.
```
Comment (guimarqu) [inline src/ColunaBase/solsandbounds.jl:46]:
```suggestion
    diff(b1, b2)

Distance between a primal bound and a dual bound that have the same objective sense.
Distance is non-positive if dual bound reached primal bound.
```

File: src/ColunaBase/solsandbounds.jl [hunk 3]
```diff
@@ -82,9 +83,15 @@ function gap(db::Bound{<:Dual,<:MaxSense}, pb::Bound{<:Primal,<:MaxSense})
...
-
-doc todo
+    printbounds(
+        db,
+        pb,
+        io
+    )
+    
+Prints the lower and upper bound taking into account the objective sense `Sense<:AbstractSense`.
+
```
Comment (guimarqu) [inline src/ColunaBase/solsandbounds.jl:93]:
```suggestion
    printbounds(db, pb [, io])
    
Prints the lower and upper bound according to the objective sense.

```

File: src/ColunaBase/solsandbounds.jl [hunk 4]
```diff
@@ -167,6 +174,12 @@ solution status should be :
     UNKNOWN_SOLUTION_STATUS, UNCOVERED_SOLUTION_STATUS
 )
 
+"""
+    convert_status
+
+Convert the termination status, i.e., `MOI.TerminationStatus` , `TerminationStatus`, to one another.
+The function also does the same thing regarding `MOI.ResultStatusCode` and `SolutionStatus`.
+"""
```
Comment (guimarqu) [inline src/ColunaBase/solsandbounds.jl:182]:
```suggestion
"""
    convert_status(status::MOI.TerminationStatusCode) -> Coluna.TerminationStatus
    convert_status(status::Coluna.TerminationStatus) -> MOI.TerminationStatusCode
    convert_status(status::MOI.ResultStatusCode) -> Coluna.SolutionStatus

Convert a termination or solution `status` of a given type to the corresponding status in another type.
This method is used to communicate between Coluna and MathOptInterface.
"""
```

File: src/ColunaBase/solsandbounds.jl [hunk 5]
```diff
@@ -207,17 +220,61 @@ struct Solution{Model<:AbstractModel,Decision,Value} <: AbstractDict{Decision,Va
...
+        status::SolutionStatus
+    )
+
+Create a solution to the Model
+
+Other arguments are: 
+- `decisions` is a Vector with the index of each decision.
+- `values` is a Vector with the values for each decision.
+- `solution_value` is the value of the solution.
+- `status` is the solution status.
```
Comment (guimarqu) [inline src/ColunaBase/solsandbounds.jl:237]:
```suggestion
Create a solution to the `model`. Other arguments are: 
- `decisions` is a vector with the index of each decision.
- `values` is a vector with the values for each decision.
- `solution_value` is the value of the solution.
- `status` is the solution status.
```

File: src/ColunaBase/solsandbounds.jl [hunk 6]
```diff
@@ -207,17 +220,61 @@ struct Solution{Model<:AbstractModel,Decision,Value} <: AbstractDict{Decision,Va
...
+"""
+    getsol(
+        s::Solution
+    )
+Parameters:
+- `s` is an instance of Solution.
+
+Return:
+- return a dynamic sparse vector that contains the solution.
+"""
```
Comment (guimarqu) [inline src/ColunaBase/solsandbounds.jl:253]:
```suggestion
"""
    getsol(solution)
    
Return the dynamic sparse vector that describes `solution`.
"""
```

File: src/ColunaBase/solsandbounds.jl [hunk 7]
```diff
@@ -207,17 +220,61 @@ struct Solution{Model<:AbstractModel,Decision,Value} <: AbstractDict{Decision,Va
...
+    getvalue(
+        s::Solution
+    )
+Parameters:
+- `s` is an instance of Solution.
+
+Return:
+- getvalue will return the Solution's bound float value.
+"""
```
Comment (guimarqu) [inline src/ColunaBase/solsandbounds.jl:265]:
```suggestion

"""
    getvalue(solution) -> Float64

Return the value of `solution`.
"""
```

File: src/ColunaBase/solsandbounds.jl [hunk 8]
```diff
@@ -207,17 +220,61 @@ struct Solution{Model<:AbstractModel,Decision,Value} <: AbstractDict{Decision,Va
...
+    Coluna.ColunaBase.getstatus(
+        s::Solution
+    )
+Parameters of Coluna.ColunaBase.getsol:
+- `s` is an instance of the Solution Struct.
+
+Return:
+- getstatus will return the Solution's status
+"""
```
Comment (guimarqu) [inline src/ColunaBase/solsandbounds.jl:277]:
```suggestion
"""
    getstatus(solution) -> SolutionStatus

Return the solution status of `solution`.
"""
```

File: src/ColunaBase/solsandbounds.jl (hunk 4 above)
Comment (ItamarRocha) [reply src/ColunaBase/solsandbounds.jl:178]: It is missing one of the convert_status functions, isn't it?

```
convert_status(coluna_status::SolutionStatus) -> MOI.ResultStatusCode
```

Comment (guimarqu) [reply src/ColunaBase/solsandbounds.jl:178]: I saw 3 methods in the file.

**Review by guimarqu — APPROVED** [review]

**Pull request merged**